hack/prepopulate-kind-registry.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
```

Comment:
This is an improvement but what I had in mind is to make a kind node image available with all docker images preloaded since that would help us save time pulling images that the proposed script is required to do.

See https://github.com/kubernetes-sigs/kind/issues/2063#issuecomment-775734522
The idea is to check in a kind node image to the repository and have the kind cluster be spun up with this node image during the CI pipeline.

Reply:
I will look into this. What I did is that kind will first try to pull images from local registry and if it didn't find one then it will pull from the remote. The main reason I did this is because if the image "Pull Policy" is not IfNotPresent, Never or tag latest, Kind will pull image from remote repository even if one is available locally (preloaded). https://iximiuz.com/en/posts/kubernetes-kind-load-docker-image/

So, what I'm thinking of doing now is load all images (using `kind load ...`) that has "Pull Policy" IfNotPresent, Never, or tag set to version (not latest) and rest from the local registry. Let me know if you have any suggestions.

Reply:
Are you suggesting you will load the images and then take a snapshot of the node image and check it in or will this need to happen each time the action is run? The former would save us time since we will not need to pull an image from the remote registry while the latter would not work because you cannot cache images across iterations on Github Actions

Reply:
@nitishm Can you have a look at the changes. I'm loading all the images that doesn't have a tag "latest" or image pull policy "Always" to kind node. And rest to local registry.